tests/test_communication.py

```diff
@@ -71,6 +71,26 @@ def test_connect_servo_ecat(read_config):
     assert ifname in mc.net and mc.net[ifname] is not None
 
 
+@pytest.mark.smoke
+@pytest.mark.soem
+def test_get_ifname_from_interface_ip(mocker, read_config):
+    ip = type('IP', (object,), {
+        'ip': '192.168.2.1',
+        'is_IPv4': True
+    })
+    adapter = type('Adapter', (object, ), {
+        'ips': [ip],
+        'name': b'{192D1D2F-C684-467D-A637-EC07BD434A63}'
+    })
+    mocker.patch('ifaddr.get_adapters', return_value=[adapter])
+    mc = MotionController()
+    soem_config = read_config["soem"]
```

Comment:
I don't understand why you get the interface_ip from the config.json file. If you change it the test will fail, it has no sense...